winlogbeat/sys/wineventlog/metadata_store.go

```diff
@@ -267,7 +267,7 @@ func (s *PublisherMetadataStore) getEventMetadata(eventID uint16, version uint8,
 	// user data from a handle, since this information is not available
 	// from the metadata or anywhere else. It is not ideal to update the defaultEM
 	// here but there is no way around it at the moment.
-	if em.EventData.IsUserData {
+	if defaultEM != nil && em.EventData.IsUserData {
 		defaultEM.EventData.IsUserData = true
 		defaultEM.EventData.Name = em.EventData.Name
 	}
```

Comment:
it seems `em` at L252 will also be null, so there's still null dereference below at L277

shouldn't this function return at L251 if `defaultEM` is null?

Reply:
we use defaultEM as a fallback at any point in the function, so even if it is null we will still try to get the metadata, and only resort to defaultEM in case something goes south, at that point, if we return a nil defaultEM it really does not matter as the function is designed so it can return a nil that caller must check

Reply:
ok, since this `em.equal(defaultEM)` is a valid statement with both variables being null, LGTM